paddle/fluid/framework/ir/graph_helper.cc

```diff
@@ -497,6 +505,70 @@ static OpDesc *ReplaceScaleLossGradOp(const Node &node, OpDesc *desc) {
   return desc;
 }
 
+static void ReplaceAllReduceOp(const Node &node,
+                               proto::BlockDesc *block,
+                               std::vector<OpDesc> *ops) {
+#if defined(PADDLE_WITH_NCCL) || defined(PADDLE_WITH_RCCL)
+  ops->emplace_back();
+  auto &desc1 = ops->back();
+  std::string name = "fake_coalesce_" + std::to_string(ops->size());
```

Comment:
Not important, the _name_ is not used until line 537 in _if_ statement, better to be constructed inside the _if_ statement.

Reply:
It is for the correct ops->size().